tests/components/habitica/test_button.py

```diff
@@ -340,3 +347,65 @@ async def test_button_unavailable(
     for entity_id in entity_ids:
         assert (state := hass.states.get(entity_id))
         assert state.state == STATE_UNAVAILABLE
+
+
+async def test_class_change(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    aioclient_mock: AiohttpClientMocker,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test removing and adding skills after class change."""
+    mage_skills = [
+        "button.test_user_chilling_frost",
+        "button.test_user_earthquake",
+        "button.test_user_ethereal_surge",
+    ]
+    healer_skills = [
+        "button.test_user_healing_light",
+        "button.test_user_protective_aura",
+        "button.test_user_searing_brightness",
+        "button.test_user_blessing",
+    ]
+    aioclient_mock.get(
+        f"{DEFAULT_URL}/api/v3/user",
+        json=load_json_object_fixture("wizard_fixture.json", DOMAIN),
+    )
+    aioclient_mock.get(
```

Comment:
in general, I don't like direct aioclient mocks in HA. Instead I'd recommend mocking the library's `client`. Since the other tests are already using this pattern, I'll allow it. Maybe you think about changing this from the entire tests in the future.

Reply:
i know i know, but that is not possible here. There's a PR that introduces a new library and all tests have been rewritten accordingly